src/GraphQL/Builders/ConnectionBuilder_Typed.cs

```diff
@@ -0,0 +1,291 @@
+using System;
+#if NETSTANDARD2_1
+using System.Diagnostics.CodeAnalysis;
+#endif
+using System.Linq;
+using System.Threading.Tasks;
+using GraphQL.Types;
+using GraphQL.Types.Relay;
+
+#nullable enable
+
+namespace GraphQL.Builders
+{
+    /// <summary>
+    /// Builds a connection field for graphs that have the specified source and return type.
+    /// </summary>
+    public class ConnectionBuilder<TSourceType, TReturnType>
+    {
+        private bool _isBidirectional => FieldType.Arguments.Any(x => x.Name == "before");
```

Comment:
What if some one add `before` field of other type ?

Reply:
It was a property before; you suggested I replace it with a calculated property, so I did so.  If you think it should be different, please suggest a change.  I don't really care and I don't think it matters.  It made more sense to me the other way.  I do like that PageSize is saved within metadata so that extension methods can create a resolver properly.

Reply:
I talked about `before` _argument_, not _field_, sorry. Now property checks only one argument by name, even not by type.

Reply:
True. Seems fine to me.

If you want to suggest different code, I’ll merge your suggestion. I’m not changing it otherwise.

Reply:
```csharp
private bool IsBidirectional => FieldType.Arguments.Find("before")?.ResolvedType is StringGraphType && FieldType.Arguments.Find("last")?.ResolvedType is IntGraphType;
```